File: chapter-1/problems/python/is_prime.py
def isPrime(n : int)-> bool:
    if n<=1:
        return False
    
    if n==2 or n==3:
        return True
    
    if n%2==0 or n%3==0:
        return False
    
    for i in range(2,ceil(sqrt(n))+1):
        if n%i==0:
            return False
    
    return True

n = int(input("Enter the value of n : "))

if(isPrime(n)):
    print(f"{n} is prime")
else:
    print(f"{n} is not prime")

File: chapter-1/problems/cpp/is_prime.cpp
#include<bits/stdc++.h>

using namespace std;

bool isPrime(int n){
    if(n<=1){
        return false;
    } 
    if( n==2 || n==3){
        return true;
    }

    if(n%2==0 || n%3==0){
        return false;
    }
    
    for(int i = 2;i<=sqrt(n);i++){
        if(n%i==0){
            return false;
        }
    }
    return true;
}

int main(){
    int n;
    cout<<"Enter a number n : ";
    cin>> n;

    if(isPrime(n)){
        cout<<"Prime"<<endl;
    }
    else{
        cout<<"Not Prime"<<endl;
    }

    return 0;
}